.github/workflows/release.yml

```
@@ -0,0 +1,36 @@
+name: 'Release'
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: write
+  id-token: write
```

there is no need to write anything and OIDC token can be added to PyPI job

```suggestion
  contents: read
```